### ISSUE TITLE
Apostrophe rendering as left smart-quote instead of right, when the possessive noun has a tooltip.

### ISSUE THREAD

**Issue by feasgal** [squidfunk/mkdocs-material#4651]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

If `Word` is defined as an abbreviation using the `*[Word]:` syntax, then `Word's` (e.g., `this Access Group's permissions`) renders with a left smart-quote instead of an apostrophe.

### Expected behaviour

I would expect this markdown source:

```
Each section of the editor controls an access group's ability to view, and actions within, the listed pages.
Select the checkboxes to allow the appropriate actions.

*[access group]: Access groups allow admin users to<br>assign custom permissions to all<br>users assigned to that group.
```

To render wie so:

![image](https://user-images.githubusercontent.com/97999520/202267595-bd6dcfae-17be-47fd-9a95-b5b6fbeedc31.png)

### Actual behaviour

It is rendering like this instead:

![image](https://user-images.githubusercontent.com/97999520/202267775-eee15f8f-7697-4a92-9e9b-eefd61999a57.png)

### Steps to reproduce

1. Put the above markdown source in a file.
2. Use the below configuration.
3. `mkdocs serve`

### Package versions

- Python: `3.10.0`
- MkDocs: `1.4.0`
- Material: `8.5.6+insiders.4.25.2`

### Configuration

```yaml
site_name: My Documentation
site_url: ""

plugins:
    - search:
        separator: '[\s\-\./\\]+'
    - offline

theme:
  name: material
  font:
    text: Montserrat
    code: Noto Sans Mono
  language: en

extra_javascript:
  - assets/mathjax/mathjax.js
  - assets/mathjax/tex-mml-chtml.js

markdown_extensions:
    - abbr: {}
    - smarty: {}
    - toc: {}
    - tables: {}
    - attr_list: {}
    - pymdownx.tabbed: {}
    - pymdownx.snippets: {}
    - meta: {}
    - md_in_html: {}
    - admonition: {}
    - pymdownx.details: {}
    - pymdownx.superfences: {}
    - def_list: {}
    - pymdownx.emoji:
        emoji_index: !!python/name:materialx.emoji.twemoji
        emoji_generator: !!python/name:materialx.emoji.to_svg
    - pymdownx.caret: {}
    - pymdownx.mark: {}
    - pymdownx.tilde: {}
    - pymdownx.keys: {}
    - footnotes: {}
    - pymdownx.arithmatex:
        generic: true
```

### System information

- Operating system: macOS Monterey 12.6
- Browser: Chrome 107.0.5304.110, Safari 15.6.1, Firefox 106.0.2

**Comment by facelessuser** [comment]
This is not a bug with Material but an issue with the `smarty` extension. Personally, I always turn off `smart_quotes` when using smarty for this and other reasons. I'm sure it could be improved though, but an issue would have to be filed upstream.

**Comment by feasgal** [comment]
For anyone else who ends up reading this, I didn't want to give up `smart_quotes`, so for now I'm fixing it with javascript. Not ideal, but needs must.

``` js
function replaceRecursively(element, from, to) {
    if (element.childNodes.length) {
        element.childNodes.forEach(child => replaceRecursively(child, from, to));
    } else {
        const cont = element.textContent;
        if (cont) element.textContent = cont.replace(from, to);
    }
};

replaceRecursively(document.body, new RegExp("‘s ", "g"), "’s ");
```

**Comment by facelessuser** [comment]
That's find as a stopgap solution, or if you just don't mind maintaining such a solution, but ideally, an issue should be created on the original extension. There are all sorts of contractions, and a more intelligent way to address them would be ideal. I'm sure more issues like this will crop up in the future.

**Comment by feasgal** [comment]
I just did, but it's based on the original SmartyPants plugin, which [specifically mentions this](https://daringfireball.net/projects/smartypants/#:~:text=or%20email%20client.-,ALGORITHMIC%20SHORTCOMINGS) as something to not bother filing a bug for. Given that, I ended up at the javascript solution for now, and if it gets a real fix in `smarty`, that will be nice, but what I found so far didn't make me super optimistic. We'll see!
